x-pack/legacy/plugins/apm/server/lib/helpers/es_client.ts

```diff
@@ -92,10 +92,23 @@ interface APMOptions {
   includeLegacyData: boolean;
 }
 
-export function getESClient(req: Legacy.Request) {
+interface ClientCreateOptions {
+  asInternalUser?: boolean;
+}
+
+export type ESClient = ReturnType<typeof getESClient>;
+
+export function getESClient(
+  req: Legacy.Request,
+  { asInternalUser = false }: ClientCreateOptions = {}
+) {
   const cluster = req.server.plugins.elasticsearch.getCluster('data');
   const query = req.query as Record<string, unknown>;
 
+  const callMethod = asInternalUser
+    ? cluster.callWithInternalUser
+    : cluster.callWithRequest.bind(cluster, req);
```

Comment:
Looking forward to partial application coming to javascript. Will enable us to do `cluster.callWithRequest(req, ?, ?)`. Until then, `bind` is our friend :)